src/lib/y2storage/proposal/autoinst_devices_creator.rb

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/proposal/partitions_distribution_calculator"
+# TODO: fix distribution calculator to don't require this
+require "y2storage/proposal/lvm_helper"
+require "y2storage/proposal/partition_creator"
+
+module Y2Storage
+  module Proposal
+    # Class to create and reuse devices during the AutoYaST proposal, based
+    # on the information contained in the profile.
+    class AutoinstDevicesCreator
```

Comment:
then it would be nice to document this different behavior. In which part it is different? it will help if e.g. me deciding which one I should use when I want to propose something.